java/java.source.base/src/org/netbeans/modules/java/source/save/CasualDiff.java

```diff
@@ -4151,7 +4164,7 @@ private int diffList(
                         match = wasInFieldGroup;
                         for (Iterator<JCTree> it = deletedItems.iterator(); !match && it.hasNext(); ) {
                             ld = it.next();
-                            if (match == treesMatch(item.element, ld, false)) {
+                            if (match = treesMatch(item.element, ld, false)) {
```

Comment:
Since this expression seems confusing, I'd recomment to make a comment note that the assignment is really meant :)

Reply:
Thanks for the input. Have made the change.